test/EndToEndTests/Tests/Client/Microsoft.OData.Client.E2E.Tests/Common/Client/Default/DefaultServiceCsdl.xml

```diff
@@ -146,9 +146,9 @@
         <Property Name="CountryRegion" Type="Edm.String" />
         <Property Name="AccountInfo" Type="Microsoft.OData.Client.E2E.Tests.Common.Server.Default.AccountInfo" />
         <Property Name="UpdatedTime" Type="Edm.DateTimeOffset" Nullable="false" />
-        <NavigationProperty Name="MyPaymentInstruments" Type="Collection(Microsoft.OData.Client.E2E.Tests.Common.Server.Default.PaymentInstrument)" />
-        <NavigationProperty Name="ActiveSubscriptions" Type="Collection(Microsoft.OData.Client.E2E.Tests.Common.Server.Default.Subscription)" />
-        <NavigationProperty Name="MyGiftCard" Type="Microsoft.OData.Client.E2E.Tests.Common.Server.Default.GiftCard" />
+        <NavigationProperty Name="MyPaymentInstruments" Type="Collection(Microsoft.OData.Client.E2E.Tests.Common.Server.Default.PaymentInstrument)" ContainsTarget="true" />

```

Comment:
> This is also a shared csdl and will be required when testing containment tests

If for this reason we change the navigation properties to be contained, what CSDL will use for non-containment tests?

Reply:
For the specific types they are meant to test containment. There are other types for testing non-containment. Also if common csdl does not have what is required then one create a new csdl with the types he or she needs to test whichever scenario they want to test.

Reply:
I don't want to appear to flog the horse to death but my point is, if you have a CSDL and a model checked into the repo, that was intended for non-contained scenario tests, you don't modify it to test contained scenario tests. What you do is introduce a new CSDL and model to test the contained scenario tests. Plus, this pull request is for complex types tests, not containment.

Reply:
Yes. I totally agree and what I'm trying to say is: This is a CSDL in a common folder. Meaning it is shared by several tests. And from the word go like in 7.x release branch it has the various types as contained but I had forgotten to include them. Just made those updates to have something similar to what was there before. So there are some common CSDLs that will be shared by various tests and these tests CSDLs have most of these things.